packages/documentation/copy/pt/reference/Utility Types.md

```diff
@@ -0,0 +1,367 @@
+---
+title: Tipos Utilitários
+layout: docs
+permalink: /pt/docs/handbook/utility-types.html
+oneline: Tipos que são inclusos globalmente em TypeScript
+translatable: true
+---
+
+TypeScript provém vários tipos utilitários para facilitar transformações de tipo comum. Esses utilitários estão disponíveis globalmente.
+
+## `Partial<Type>`
+
+Constroi um tipo com todas as propriedades de `Type` definidas como opcional. Esse utilitário irá retornar um tipo que representa todos os subconjuntos de um determinado tipo.
+
+##### Exemplo
+
+```ts twoslash
+interface Todo {
+  titulo: string;
+  descricao: string;
+}
+
+function atualizaTodo(todo: Todo, camposParaAtualizar: Partial<Todo>) {
+  return { ...todo, ...camposParaAtualizar };
+}
+
+const todo1 = {
+  titulo: "organizar a mesa",
+  descricao: "limpar bagunça",
+};
+
+const todo2 = atualizaTodo(todo1, {
+  descricao: "tirar o lixo",
+});
+```
+
+## `Readonly<Type>`
+
+Constroi um tipo com todas as propriedades de `Type` definidas como `readonly`, significando que as propriedades do tipo construído não podem ser reatribuídas.
+
+##### Exemplo
+
+```ts twoslash
+// @errors: 2540
+interface Todo {
+  titulo: string;
+}
+
+const todo: Readonly<Todo> = {
+  titulo: "Deleta usuários inativos",
+};
+
+todo.titulo = "Olá";
+```
+Esse utilitário é útil para representar expressões de atribuição que irão falhar em tempo de execução (Ex. Ao tentar reatribuir propriedades de um [objeto congelado](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze)).
+
+##### `Object.freeze`
+
+```ts
+function freeze<Type>(obj: Type): Readonly<Type>;
+```
+
+## `Record<Keys,Type>`
+
+Constroi um tipo com um conjunto de propriedades `Keys` do tipo `Type`. Esse utilitário pode ser usado para mapear as propriedades de um tipo para outro tipo.
+
+##### Exemplo
+
+```ts twoslash
+interface InfoPagina {
+  titulo: string;
+}
+
+type Pagina = "inicio" | "sobre" | "contato";
+
+const nav: Record<Pagina, InfoPagina> = {
+  sobre: { titulo: "sobre" },
+  contato: { titulo: "contato" },
+  home: { titulo: "inicio" },
+};
+
+nav.about;
```

Comment:
You left `home` and `about` in this example, which is what the compiler is complaining about. I believe you want:

```suggestion
  inicio: { titulo: "inicio" },
};

nav.sobre;
```

Reply:
That's it ! Thanks @orta 😄